sim-lib/src/sim_node.rs

```diff
@@ -291,6 +310,21 @@ impl SimulatedChannel {
         }
     }
 
+    /// Validates that a simulated channel has distinct node pairs and valid routing policies.
+    fn validate(&self) -> Result<(), SimulationError> {
+        if self.node_1.policy.pubkey == self.node_2.policy.pubkey {
+            return Err(SimulationError::SimulatedNetworkError(format!(
+                "Channel should have distinct node pubkeys, got: {} for both nodes.",
+                self.node_1.policy.pubkey
+            )));
+        }
+
+        self.node_1.policy.validate(self.capacity_msat)?;
+        self.node_2.policy.validate(self.capacity_msat)?;
+
```

Comment:
Would it be worth also validating that the `ChannelState`'s have valid local balances here? It appears that you could create a `SimulatedChannel` with a total capacity that is different than `node_1.local_balance_msat + node_2.local_balance_msat`? This is caught in a sanity check later, but if the goal of this PR is to catch things earlier, this could be an easy one to check.

```rust
if self.node_1.local_balance_msat + self.node_2.local_balance_msat != self.capacity_msat {
    return Err(SimulationError::SimulatedNetworkError(format!(
        "Local balances (node1: {}, node2: {}) do not equal the total channel capacity: {}",
        self.node_1.local_balance_msat,
        self.node_2.local_balance_msat,
        self.capacity_msat
    )));            
}
```
This does not take into account in-flight sats, but if we are validating the creation of this channel that might not matter?

Reply:
Yeah that's great! 

I think we can take it further and not allow a non-zero in-flight balance, since as-is we'll never be able to clear out that in-flight balance if you start with it. Sure we _might_ get a use case for that one day, but start strict + follow complaint-driven-develpoment sgtm.

Pushed the above + a (quite lazy) test